include/proteus/Frontend/Func.hpp

```diff
@@ -151,6 +153,13 @@ class FuncBase {
     return Lower(*this);
   }
 
+  ForLoopBuilder forLoop(LoopBoundsDescription Bounds);
+  ForLoopBuilder forLoop(LoopBoundsDescription Bounds,
+                         std::function<void()> Body);
+
+  LoopNestBuilder buildLoopNest(std::vector<ForLoopBuilder> Loops);
```

Comment:
Instead of a vector I'd use a variadic template to gather the `ForLoopBuilder` templated on lambda objects.

Ping me if that doesn't make sense. 

We may do such refactoring at a later time but it makes time to do it now that you have the code fresh in your mind